### PR TITLE
mgmt: mcumgr: Add iterable section to register MCUmgr handlers

### PR DESCRIPTION
**Pull request by nordicjm**
mgmt: mcumgr: Add iterable section to register MCUmgr handlers
    
    This replaces the requirement for applications to manually
    register MCUmgr handlers by having an iterable section which
    then automatically registers the handlers at boot time.

- [X] Update code
- [X] Update samples
- ~~Documentation update~~ *There is no documentation about creating handlers, we should probably add this some time*
- [X] Release notes update
- [X] Rework transports
- [x] Re-think bluetooth registration due to how mesh sample works

### CODE REVIEW

**Comment by nordicjm** [comment]
Should probably make all the registration functions static too

File: subsys/mgmt/mcumgr/transport/Kconfig.bluetooth [hunk 1]
```diff
@@ -87,4 +87,24 @@ config MCUMGR_SMP_BT_CONN_PARAM_CONTROL_RETRY_TIME
 
 endif # MCUMGR_SMP_BT_CONN_PARAM_CONTROL
 
+config MCUMGR_SMP_BT_AUTOMATIC_INIT
```
Comment (de-nordic) [inline subsys/mgmt/mcumgr/transport/Kconfig.bluetooth:90]: Please change to `MCUMGR_TRASNPORT_<BT|UDP|UART>...` to have similar naming as proposed here https://github.com/zephyrproject-rtos/zephyr/pull/52761.

**Review by de-nordic — COMMENTED** [review]
Comment regarding Kconfig options naming, otherwise looks OK.

File: subsys/mgmt/mcumgr/transport/Kconfig.bluetooth (hunk 1 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/transport/Kconfig.bluetooth]: Fixed

**Review by de-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**